test/validations.test.js

```diff
@@ -322,6 +322,43 @@ describe('validations', function() {
       });
     });
 
+    it('should pass options to custom synch', function(done) {
+      delete User.validations;
+      User.validate('name', function(err, options) {
+        if (options.options !== 'someValue') err();
```

Comment:
`options.options` looks rather odd to me. I am proposing to rename `options.options` to something like `options.testFlag`.

If `options` object contains unexpected values, then this way of signaling the error will produce a message that's not very helpful. Unfortunately it looks like we don't have any control over the Error instance produced by our validation framework. In which case I am proposing to at least print a `console.error` message.

For example:

```js
if (options.testFlag !== 'someValue') {
  console.error(
    'Unexpected validation options: %j Expected %j',
    options, {testFlag: 'someValue'});
  err();
}
```